drivers/adc/ad4170/ad4170.c

```diff
@@ -0,0 +1,2123 @@
+				if (ret < 0)
+					return ret;
+
+				timeout = dev->rdy_conv_timeout;
+				while (rdyb == NO_OS_GPIO_LOW && timeout--) {
+					no_os_gpio_get_value(dev->gpio_dig_aux1, &rdyb);
+				};
+				if (!timeout)
+					return -ETIMEDOUT;
+
+				p++;
+				nb_samples--;
+			} while(repeat-- && nb_samples);
+
+			if (!nb_samples)
+				break;
+		}
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Read multiple samples using direct register access from Data_16b
+ * @param dev - The device structure.
+ * @param pbuf - Output pointer where to store the data values.
+ * @param nb_samples - The number of samples to read.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad4170_read16(struct ad4170_dev *dev, uint32_t *pbuf,
+		      uint16_t nb_samples)
+{
+	return _ad4170_read(dev, pbuf, nb_samples, AD4170_REG_DATA_16b);
+}
+
+/**
+ * @brief Read multiple samples using direct register access from Data_16b_Status
+ * @param dev - The device structure.
+ * @param pbuf - Output pointer where to store the data values.
+ * @param nb_samples - The number of samples to read.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad4170_read16s(struct ad4170_dev *dev, uint32_t *pbuf,
+		       uint16_t nb_samples)
+{
+	return _ad4170_read(dev, pbuf, nb_samples, AD4170_REG_DATA_16b_STATUS);
+}
+
+/**
+ * @brief Read multiple samples using direct register access from Data_24
+ * @param dev - The device structure.
+ * @param pbuf - Output pointer where to store the data values.
+ * @param nb_samples - The number of samples to read.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad4170_read24(struct ad4170_dev *dev, uint32_t *pbuf,
+		      uint16_t nb_samples)
+{
+	return _ad4170_read(dev, pbuf, nb_samples, AD4170_REG_DATA_24b);
+}
+
+/**
+ * @brief Read multiple samples using direct register access from Data_24_Status
+ * @param dev - The device structure.
+ * @param pbuf - Output pointer where to store the data values.
+ * @param nb_samples - The number of samples to read.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad4170_read24s(struct ad4170_dev *dev, uint32_t *pbuf,
+		       uint16_t nb_samples)
+{
+	return _ad4170_read(dev, pbuf, nb_samples, AD4170_REG_DATA_24b_STATUS);
+}
+
+/**
+ * @brief Read multiple samples using direct register access from Data_32b
+ * @param dev - The device structure.
+ * @param pbuf - Output pointer where to store the data values.
+ * @param nb_samples - The number of samples to read.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad4170_read32(struct ad4170_dev *dev, uint32_t *pbuf,
+		      uint16_t nb_samples)
+{
+	return _ad4170_read(dev, pbuf, nb_samples, AD4170_REG_DATA_32b);
+}
+
+static int32_t _scratchpad_test(struct ad4170_dev *dev, uint8_t pattern)
+{
+	int32_t ret;
+	uint32_t reg;
+
+	ret = ad4170_spi_reg_write(dev, AD4170_REG_SCRATCH_PAD, pattern);
+	if (ret < 0)
```

Comment:
same goes for `if (ret)` convention linux compliant.

Reply:
will do